ecosystem/typescript/sdk_v2/src/types/account_address.ts

```diff
@@ -0,0 +1,241 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import { bytesToHex, hexToBytes } from "@noble/hashes/utils";
+import { Serializer, Deserializer, Bytes } from "../bcs";
+
+// TODO: Get feedback on these error messages, ideally they can be used directly in
+// dapps and wallets. The enum is meant to make it possible for daps to user their
+// own error messages if they want, but the defaults should still be good, and they're
+// not great right now.
+
+// These options influence fromStr and the isValid* functions. You may use them to make
+// the parsing of those functions strictier. Not setting these options is the best way
+// to comply with AIP 40 but while migrating it may be helpful to be able to enforce
+// stricter behavior. If we agree to make these two enabled by default, I can update
+// the AIP to say that this is what people SHOULD do but they MAY opt in to accepting
+// addresses more leniently in specifically these two ways.
+export type ValidityOptions = {
+  // Only addresses in the long form (64 characters + optionally a leading 0x) will be
+  // considered valid unless the address is special.
+  requireLongFormUnlessSpecial?: boolean;
+
+  // Only addresses in the long form (64 characters + optionally a leading 0x) will be
+  // considered valid, even if the address is special.
+  requireLongForm?: boolean;
+
+  // If set, the address string will be considered invalid if there is no leading 0x.
+  requireLeadingZeroX?: boolean;
+};
+
+// TODO: As part of reviewing this we should discuss these options. I'm partially
```

Comment:
That's a fair point. So here's what I want to update the standard to:

Functions that parse / validate strings as AccountAddress MUST accept the following:

a. LONG
b. SHORT for special addresses

The following correspond to representations actively in circulation. Implementations MAY accept these formats for the sake of backwards compatibility with the existing ecosystem.

c. LONG with or without leading 0x
d. SHORT with or without leading 0x for all addresses (not just special)

These are allowed for the sake of migration, since some tools return account addresses in these latter formats and cannot be updated without breaking backwards compatibility. Where possible, developers should avoid accepting these formats (e.g. when building APIs, wallets, tools, etc) but it is allowed in order to maintain compatibility.

Reply:
We spoke offline, I'll make a change to the AIP.